### PR TITLE
Remove docker compose

### PR DESCRIPTION
**Pull request by ross-p-smith**
# PR for issue #1135

## What is being addressed
Move to a devcontainer.json rather than a compose.yml

## How is this addressed
I've also added default values to the Dockerfile

### CODE REVIEW

**Review by ross-p-smith — COMMENTED** [review]
@marrobi - We might have to work together and add some variables to the build to remove this

File: .github/workflows/deploy_tre.yml [hunk 1]
```diff
@@ -51,6 +46,7 @@ jobs:
       - name: Deploy TRE
         shell: bash
         env:
+          TF_IN_AUTOMATION: 1
```
Comment (marrobi) [inline .github/workflows/deploy_tre.yml:49]: Should this be part of this PR?

Comment (ross-p-smith) [reply .github/workflows/deploy_tre.yml:49]: Yes, if you can add these as secrets in your build,       -e ARM_CLIENT_ID \
      -e ARM_CLIENT_SECRET \
      -e ARM_TENANT_ID \
      -e ARM_SUBSCRIPTION_ID \

Comment (marrobi) [reply .github/workflows/deploy_tre.yml:49]: What's this got to do with the docker compose file? Thought this was a separate issue regarding amending the deployment scripts?

Comment (martinpeck) [reply .github/workflows/deploy_tre.yml:49]: Hey @marrobi. I had a chat with @ross-p-smith and this change isn't part of the move away from Compose so he'll remove this file from the PR.

**Review by martinpeck — CHANGES_REQUESTED** [review]
Once `deploy_tre.yml` is removed, I'm happy with the rest of the PR.

File: .github/workflows/deploy_tre.yml (hunk 1 above)
Comment (ross-p-smith) [reply .github/workflows/deploy_tre.yml]: I was trying to be nice, but I will fix up GH Actions as a separate task

**Comment by ross-p-smith** [comment]
@marrobi and @martinpeck - I've added the capability to run the container as a non root user into this PR. (Also removed the GH Action change)

**Review by martinpeck — APPROVED** [review]

**Pull request merged**